src/psyclone/tests/psyir/frontend/fparser2_fortran_use_test.py

```diff
@@ -49,6 +49,33 @@
     DataSymbol, AutomaticInterface, INTEGER_SINGLE_TYPE
 
 
+def test_use_return(parser):
+    ''' Check the the Fparser correctly handles when a function uses a
+    return variable with a kind defined inside the function.'''
+    processor = Fparser2Reader()
+    reader = FortranStringReader("real(rkind) function x()\n"
+                                 "  use my_mod, only: rkind\n"
+                                 "  x = 1.0_rkind\n"
+                                 "end function x\n")
+    ptree = parser(reader)
+    processor.generate_psyir(ptree)
```

Comment:
Could you query the resulting symbol, e.g.:

    psyir = ...
    sym = psyir.children[0].symbol_table.lookup("x")
    assert isinstance(sym, RoutineSymbol)
    assert sym.datatype is ...

Reply:
Added a few checks to both tests.
One thing I'm a little unclear on from your suggestion vs what I found is that the `RoutineSymbol` for `x` doesn't seem to exist when the `Routine` exists outside of a `Module` - if its only contained in the `FileContainer` then the `RoutineSymbol` isn't store in a `SymbolTable` anywhere.

Reply:
Oh yes, I forgot that only a Container has a SymbolTable, a FileContainer doesn't. That's OK.